CHANGELOG.md

```diff
@@ -1,7 +1,39 @@
 # Changelog
 
-## Version 0.38.0
-_In development_
+## Version 0.38.0 
+
+([Full Changelog](https://github.com/unitaryfund/mitiq/compare/v0.37.0...v0.38.0))
```

Comment:
Not 100% needed, but this does cause the link check test to fail since this is not yet an existing URL. Check out how Alessandro did it last release in https://github.com/unitaryfund/mitiq/commit/32b53c2e29d1b96d1b17077e4a219ef3a269cd0d. You will have to edit the release notes when doing the release, so you can uncomment it then.

Again, not totally needed, but it is nice to try to have CI pass before merging if possible.